frontend/app/electron/main/subprocess-handler.ts

```diff
@@ -167,6 +176,68 @@ export class SubprocessHandler {
     this.coreManager.start(command, args, workDir);
   }
 
+  /**
+   * Checks the availability of the core API by sending periodic ping requests
+   * to the ping endpoint.
+   * Continues to retry pinging the URL for a specified number of attempts,
+   * applying a delay between each attempt if the initial request fails.
+   *
+   * @param {string} url - The base URL of the core API endpoint to ping.
+   * @param {number} [retries=10] - The maximum number of ping attempts before giving up.
+   * @param {number} [waitSeconds=6] - The delay, in seconds, between consecutive ping attempts.
+   * @return {Promise<boolean>} A promise that resolves to true if the API responds successfully within the given attempts, otherwise false.
+   */
+  private async checkCoreApiAvailability(
+    url: string,
+    retries: number = 10,
+    waitSeconds: number = 6,
+  ): Promise<boolean> {
+    await wait(3000);
```

Comment:
why this wait of 3 secs? Let the app enough time for the app to run? Shouldn't it be less?

Reply:
Yes so if the code runs immediately the backend won't give a proper response because it is close to spawning, so it would have to wait a bit. 

When I was trying locally it was around 2ish seconds so 3 just to be in time.

That being said, there is a problem that we need to think about both this and the docker and this is when do we decide to bail out.